streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -223,14 +203,18 @@ public String queryableStoreName() {
 
         builder.addGraphNode(this.streamsGraphNode, tableNode);
 
+        // we preserve the key following the order of 1) materialized, 2) parent, 3) null
+        // we preserve the value following the order of 1) materialized, 2) null
```

Comment:
I think these comments would be more helpful if they explained the exceptions.

It is generally true on any DSL method that an explicitly provided serde overrides the one passed from the parent.

By the way, since the serdes are nullable, null is always the last option (as in the comment on L132, maybe it doesn't need to be stated).

For example, in this case, we could simply say `// don't inherit parent value serde, since this operation may change the value type`.

Reply:
I want to specifically state the precedence ordering if the serde is explicitly specified from multiple places. But I agree it's better to add the explanation where we would not inherit from parent.

Reply:
Cf my comment above.